homeassistant/components/sensor/hmz19.py

```diff
@@ -0,0 +1,79 @@
+"""
+Support for CO2 sensor connected to a serial port.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.serial_pm/
+"""
+import logging
+import voluptuous as vol
+
+from homeassistant.const import CONF_NAME
+from homeassistant.helpers.entity import Entity
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+
+REQUIREMENTS = ['pmsensor==0.3']
+
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_SERIAL_DEVICE = "serial_device"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_NAME, default=""): cv.string,
+    vol.Required(CONF_SERIAL_DEVICE): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the available CO2 sensors."""
+    if config.get("name") != "":
+        name = config.get("name")
+    else:
+        name = "CO2"
+        dev = HMZ19Sensor(config.get(CONF_SERIAL_DEVICE), name)
+
+    add_devices([dev])
```

Comment:
Thanks to `voluptuous` this can be re-written to a one-liner:

```python
add_devices([HMZ19Sensor(config.get(CONF_SERIAL_DEVICE), config.get(CONF_NAME))])
```

or to the style I'm using for the migration

```python
name = config.get(CONF_NAME)
device = config.get(CONF_SERIAL_DEVICE)

add_devices([HMZ19Sensor(device, name)])
```